chapter06/ard/ex1.pl

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use autodie;
+
+use 5.010;
+
+use Storable;
+
+my %data = ();
+my $datafile = "ex1.data";
+
+if ( -e -f $datafile ) {
```

Comment:
When you stack file tests, they're applied right-to-left, so the `-e` doesn't appear to do anything, since it'll only be reached if `-f` returns true.